### PR TITLE
Add curve to recipe

### PR DESCRIPTION
**Pull request by N0xMare**

### CODE REVIEW

File: contracts/Recipes/RecipeV3.sol [hunk 1]
```diff
@@ -108,6 +109,7 @@ contract RecipeV3 is IRecipe, Ownable {
     unitOracle oracle = unitOracle(0xb27308f9F90D607463bb33eA1BeBb41C27CE5AB6);
     uniV3Router uniRouter = uniV3Router(0xE592427A0AEce92De3Edee1F18E0157C05861564);
     IUniRouter sushiRouter = IUniRouter(0x1b02dA8Cb0d097eB8D57A175b88c7D8b47997506);
+    ICurveSwaps curveRouter = ICurveSwaps(0xD1602F68CC7C4c7B59D686243EA35a9C73B0c6a2);
```
Comment (clabby) [inline contracts/Recipes/RecipeV3.sol]: This address looks like Curve's mainnet `Swaps` contract, it should be the Polygon address.

File: contracts/Recipes/RecipeV3.sol [hunk 2]
```diff
@@ -397,7 +411,20 @@ contract RecipeV3 is IRecipe, Ownable {
...
+        //GET CURVE PRICE
+        try curveRouter.get_best_rate(_assetIn, _assetOut, _amountOut) returns (address memory _bestPool, uint256 amountOut) {
+            bestPool = _bestPool;
+            curveAmount = amountOut; //expected amount received in swap
+        } catch {
+            curveAmount = type(uint256).max;
+        }
+        if(bestPrice.price > curveAmount) {
+            bestPrice.price = curveAmount;
+            bestPrice.ammIndex = 5;
```
Comment (clabby) [inline contracts/Recipes/RecipeV3.sol]: The AMM index for Curve is `3` in `dexSwap`. Why is it `5` here?

**Review by clabby — CHANGES_REQUESTED** [review]

File: contracts/Recipes/RecipeV3.sol [hunk 3]
```diff
@@ -397,7 +411,20 @@ contract RecipeV3 is IRecipe, Ownable {
                 bestPrice.price = balancerAmount;
                 bestPrice.ammIndex = 4;
             } 
-        }  
+        }
+
+        //GET CURVE PRICE
+        try curveRouter.get_best_rate(_assetIn, _assetOut, _amountOut) returns (address memory _bestPool, uint256 amountOut) {
+            bestPool = _bestPool;
```
Comment (TotalPizza) [inline contracts/Recipes/RecipeV3.sol]: Why are we saving this variable `bestPool`?
I don't think I see it being used anywhere.

Comment (N0xMare) [reply contracts/Recipes/RecipeV3.sol]: Removed :+1:

**Pull request merged**